net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -0,0 +1,269 @@
+
+package net.solarnetwork.node.weather.ibm.wc;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import org.joda.time.LocalTime;
+import com.fasterxml.jackson.databind.JsonNode;
+import net.solarnetwork.node.domain.GeneralDayDatum;
+import net.solarnetwork.node.support.JsonHttpClientSupport;
+
+public class BasicWCClient extends JsonHttpClientSupport implements WCClient {
+
+	/** The default value for the {@code baseUrl} property. */
+	public static final String DEFAULT_BASE_URL = "https://api.weather.com";
+
+	/** The daily forecast url format */
+	public static final String DEFAULT_DAILY_FORECAST_URL = "v3/wx/forecast/daily/%s";
+
+	/** The default output format */
+	public static final String DEFAULT_FORMAT = "format=json";
+
+	/** The default value for the {@code locationTemplate} property */
+	public static final String DEFAULT_LOCATION_TEMPLATE = "icaoCode=%s";
+
+	/** The hourly forecast url format */
+	public static final String DEFAULT_HOURLY_FORECAST_URL = "v3/wx/forecast/hourly/%s";
+
+	/** The default value for the {@code apiKeyTemplate} property. */
+	public static final String DEFAULT_API_KEY_TEMPLATE = "apiKey=%s";
+
+	/** The default value for the {@code languageTemplate} property. */
+	public static final String DEFAULT_LANGUAGE_TEMPLATE = "language=en-US";
+
+	/** The default value for the {@code units} property. */
+	public static final String DEFAULT_UNITS = "units=e";
```

Comment:
These `DEFAULT` properties and their associated configuration class properties seem odd: the _value_ of each property should be configurable, but here both the Rest API query names and values must be configured, and must then know to include the `=` delimiter in their configuration. Is there a reason for this, like to the API query names themselves change? I was expecting something more along the lines of this:

```java
	/** The hourly forecast url format */
	public static final String DEFAULT_HOURLY_FORECAST_URL_TEMPLATE
		= "v3/wx/forecast/hourly/%s?icaoCode=%s&language=%s&format=%s&units=%s&apiKey=%s";

	/** The default output format */
	public static final String DEFAULT_FORMAT = "json";

	/** The default value for the {@code languageTemplate} property. */
	public static final String DEFAULT_LANGUAGE = "en-US";

	/** The default value for the {@code units} property. */
	public static final String DEFAULT_UNITS = "e";
```